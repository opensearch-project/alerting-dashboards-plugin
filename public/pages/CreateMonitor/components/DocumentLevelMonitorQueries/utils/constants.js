/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { OPERATORS_MAP } from '../../MonitorExpressions/expressions/utils/constants';

export const DOC_LEVEL_INPUT_FIELD = 'doc_level_input';

/**
 * A list of the operators currently supported for defining queries through the UI.
 */
export const SUPPORTED_DOC_LEVEL_QUERY_OPERATORS = [
  OPERATORS_MAP.IS.value,
  OPERATORS_MAP.IS_NOT.value,
  OPERATORS_MAP.IS_GREATER.value,
  OPERATORS_MAP.IS_GREATER_EQUAL.value,
  OPERATORS_MAP.IS_LESS.value,
  OPERATORS_MAP.IS_LESS_EQUAL.value,
];

/**
 * Initial implementation of document level monitors used these feature-specific operator values.
 * Refactored document level monitors assets to reuse the OPERATORS_MAP to reduce duplicate code
 * as new query types become supported.
 */
export const LEGACY_QUERY_OPERATORS = {
  IS: { text: 'is', value: '==' },
  IS_NOT: { text: 'is not', value: '!=' },
};

/**
 * These patterns delineate the field being queries from the query value
 * when using query string query syntax.
 */
export const QUERY_STRING_QUERY_OPERATORS = {
  [OPERATORS_MAP.IS_GREATER.value]: ':>',
  [OPERATORS_MAP.IS_GREATER_EQUAL.value]: ':>=',
  [OPERATORS_MAP.IS_LESS.value]: ':<',
  [OPERATORS_MAP.IS_LESS_EQUAL.value]: ':<=',
};

/**
 * Similar to OPERATORS_QUERY_MAP, this const contains the query patterns
 * for the SUPPORTED_DOC_LEVEL_QUERY_OPERATORS.
 */
export const DOC_LEVEL_QUERY_MAP = {
  [LEGACY_QUERY_OPERATORS.IS.value]: {
    query: ({ field, query }) => `${field}:\"${query}\"`,
  },
  [LEGACY_QUERY_OPERATORS.IS_NOT.value]: {
    query: ({ field, query }) => `NOT (${field}:\"${query}\")`,
  },
  [OPERATORS_MAP.IS.value]: {
    query: ({ field, query }) => `${field}:\"${query}\"`,
  },
  [OPERATORS_MAP.IS_NOT.value]: {
    query: ({ field, query }) => `NOT (${field}:\"${query}\")`,
  },
  [OPERATORS_MAP.IS_GREATER.value]: {
    query: ({ field, query }) =>
      `${field}${QUERY_STRING_QUERY_OPERATORS[OPERATORS_MAP.IS_GREATER.value]}${query}`,
  },
  [OPERATORS_MAP.IS_GREATER_EQUAL.value]: {
    query: ({ field, query }) =>
      `${field}${QUERY_STRING_QUERY_OPERATORS[OPERATORS_MAP.IS_GREATER_EQUAL.value]}${query}`,
  },
  [OPERATORS_MAP.IS_LESS.value]: {
    query: ({ field, query }) =>
      `${field}${QUERY_STRING_QUERY_OPERATORS[OPERATORS_MAP.IS_LESS.value]}${query}`,
  },
  [OPERATORS_MAP.IS_LESS_EQUAL.value]: {
    query: ({ field, query }) =>
      `${field}${QUERY_STRING_QUERY_OPERATORS[OPERATORS_MAP.IS_LESS_EQUAL.value]}${query}`,
  },
};
