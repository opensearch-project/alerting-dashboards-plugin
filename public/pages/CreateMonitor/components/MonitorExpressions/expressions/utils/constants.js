/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

export const POPOVER_STYLE = { zIndex: '200' };
export const EXPRESSION_STYLE = { padding: '20px', whiteSpace: 'nowrap' };
export const Expressions = {
  THRESHOLD: 'THRESHOLD',
  WHEN: 'WHEN',
  OF_FIELD: 'OF_FIELD',
  OVER: 'OVER',
  FOR_THE_LAST: 'FOR_THE_LAST',
  WHERE: 'WHERE',
  METRICS: 'METRICS',
  GROUP_BY: 'GROUP_BY',
};
export const NUMBER_TYPES = [
  'long',
  'integer',
  'short',
  'byte',
  'double',
  'float',
  'half_float',
  'scaled_float',
];
export const UNITS_OF_TIME = [
  { value: 'm', text: 'minute(s)' },
  { value: 'h', text: 'hour(s)' },
  { value: 'd', text: 'day(s)' },
];

export const WHERE_BOOLEAN_FILTERS = [
  { text: 'Select value', value: '' },
  { text: 'True', value: true },
  { text: 'False', value: false },
];

export const OPERATORS_MAP = {
  IS: {
    text: 'is',
    value: 'is',
    dataTypes: ['number', 'text', 'keyword', 'boolean'],
  },
  IS_NOT: {
    text: 'is not',
    value: 'is_not',
    dataTypes: ['number', 'text', 'keyword', 'boolean'],
  },
  IS_NULL: {
    text: 'is null',
    value: 'is_null',
    dataTypes: ['number', 'text', 'keyword', 'boolean'],
  },
  IS_NOT_NULL: {
    text: 'is not null',
    value: 'is_not_null',
    dataTypes: ['number', 'text', 'keyword'],
  },
  IS_GREATER: {
    text: 'is greater than',
    value: 'is_greater',
    dataTypes: ['number'],
  },
  IS_GREATER_EQUAL: {
    text: 'is greater than equal',
    value: 'is_greater_equal',
    dataTypes: ['number'],
  },
  IS_LESS: {
    text: 'is less than',
    value: 'is_less',
    dataTypes: ['number'],
  },
  IS_LESS_EQUAL: {
    text: 'is less than equal',
    value: 'is_less_equal',
    dataTypes: ['number'],
  },
  STARTS_WITH: {
    text: 'starts with',
    value: 'starts_with',
    dataTypes: ['text', 'keyword'],
  },
  ENDS_WITH: {
    text: 'ends with',
    value: 'ends_with',
    dataTypes: ['text', 'keyword'],
  },
  CONTAINS: {
    text: 'contains',
    value: 'contains',
    dataTypes: ['text', 'keyword'],
  },
  DOES_NOT_CONTAINS: {
    text: 'does not contain',
    value: 'does_not_contains',
    dataTypes: ['text'],
  },
  IN_RANGE: {
    text: 'is in range',
    value: 'in_range',
    dataTypes: ['number'],
  },
  NOT_IN_RANGE: {
    text: 'is not in range',
    value: 'not_in_range',
    dataTypes: ['number'],
  },
};

export const OVER_TYPES = [{ value: 'all documents', text: 'all documents' }];

export const AGGREGATION_TYPES = [
  { value: 'avg', text: 'average()' },
  { value: 'count', text: 'count()' },
  { value: 'sum', text: 'sum()' },
  { value: 'min', text: 'min()' },
  { value: 'max', text: 'max()' },
];

export const GROUP_BY_ERROR = 'Must specify at least 1 group by expression.';
export const QUERY_TYPE_GROUP_BY_ERROR = 'Can have a maximum of 1 group by selections.';

export const QUERY_TYPE_METRIC_ERROR = 'Can have a maximum of 1 metric selections.';

export const MAX_NUM_QUERY_LEVEL_GROUP_BYS = 1;
export const MAX_NUM_BUCKET_LEVEL_GROUP_BYS = 2;
export const MAX_NUM_WHERE_EXPRESSION = 1;
export const WHERE_FILTER_ALLOWED_TYPES = ['number', 'text', 'keyword', 'boolean'];
