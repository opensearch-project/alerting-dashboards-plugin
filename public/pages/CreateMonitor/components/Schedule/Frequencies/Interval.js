/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { EuiFlexItem, EuiFlexGroup } from '@elastic/eui';

import { FormikFieldNumber, FormikSelect } from '../../../../../components/FormControls';
import {
  isInvalid,
  hasError,
  validatePositiveInteger,
  validateUnit,
} from '../../../../../utils/validate';

export const unitOptions = [
  { value: 'MINUTES', text: 'Minute(s)' },
  { value: 'HOURS', text: 'Hours' },
  { value: 'DAYS', text: 'Days' },
];

export const unitToLabel = unitOptions.reduce(
  (acc, cur) => ({
    ...acc,
    [cur.value]: cur.text,
  }),
  {}
);

const Interval = () => (
  <EuiFlexGroup alignItems="flexStart" gutterSize="m">
    <EuiFlexItem>
      <FormikFieldNumber
        name="period.interval"
        formRow
        fieldProps={{ validate: validatePositiveInteger }}
        rowProps={{
          label: 'Run every',
          isInvalid,
          error: hasError,
        }}
        inputProps={{
          icon: 'clock',
          min: 1,
          'data-test-subj': 'interval_interval_field',
        }}
      />
    </EuiFlexItem>
    <EuiFlexItem>
      <FormikSelect
        name="period.unit"
        formRow
        fieldProps={{ validate: validateUnit }}
        rowProps={{
          hasEmptyLabelSpace: true,
          isInvalid,
          error: hasError,
        }}
        inputProps={{
          options: unitOptions,
          'data-test-subj': 'interval_unit_field',
        }}
      />
    </EuiFlexItem>
  </EuiFlexGroup>
);

export default Interval;
