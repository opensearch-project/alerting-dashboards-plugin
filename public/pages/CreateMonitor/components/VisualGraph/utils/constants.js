/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

/*
 *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *   Licensed under the Apache License, Version 2.0 (the "License").
 *   You may not use this file except in compliance with the License.
 *   A copy of the License is located at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *   or in the "license" file accompanying this file. This file is distributed
 *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 *   express or implied. See the License for the specific language governing
 *   permissions and limitations under the License.
 */

export const SIZE_RANGE = [3, 3];
export const LINE_STYLES = { strokeWidth: 1 };
export const ANNOTATION_STYLES = { strokeWidth: 1, stroke: 'red' };
export const HINT_STYLES = {
  background: '#3a3a48',
  borderRadius: '5px',
  padding: '7px 10px',
  fontSize: '12px',
  fontWeight: 'bold',
  color: '#fff',
  textAlign: 'left',
  whiteSpace: 'nowrap',
  boxShadow: '0 2px 4px rgba(0,0,0,.5)',
};

// Don't want values to reach top of graph so increase by a buffer % so we always have a yDomain with a slightly higher max
export const Y_DOMAIN_BUFFER = 1.4; // 40%

export const X_DOMAIN_BUFFER = 0.2; // 40%

// Size of circle for each point on graph
export const DEFAULT_MARK_SIZE = 3;

export const BAR_PERCENTAGE = 0.015;

export const BAR_KEY_COUNT = 3;
