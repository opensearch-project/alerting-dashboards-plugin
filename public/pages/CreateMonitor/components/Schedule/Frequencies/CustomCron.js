/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { EuiText, EuiLink, EuiFlexGroup, EuiFlexItem } from '@elastic/eui';

import FormikTextArea from '../../../../../components/FormControls/FormikTextArea/FormikTextArea';
import TimezoneComboBox from './TimezoneComboBox';
import { isInvalid, hasError } from '../../../../../utils/validate';
import { URL } from '../../../../../../utils/constants';

const cronHelpLink = (
  <EuiLink target="_blank" href={URL.DOCUMENTATION} external>
    cron expressions
  </EuiLink>
);

const cronHelpText = <EuiText size="s">Use {cronHelpLink} for complex schedules</EuiText>;

const CustomCron = () => (
  <EuiFlexGroup direction="column" style={{ marginTop: '5px' }}>
    <EuiFlexItem style={{ marginTop: '0px' }}>
      <FormikTextArea
        name="cronExpression"
        formRow
        rowProps={{
          label: 'Run every',
          helpText: cronHelpText,
          isInvalid,
          error: hasError,
          style: { marginTop: '0px' },
        }}
      />
    </EuiFlexItem>
    <EuiFlexItem style={{ marginTop: '0px' }}>
      <TimezoneComboBox />
    </EuiFlexItem>
  </EuiFlexGroup>
);

export default CustomCron;
