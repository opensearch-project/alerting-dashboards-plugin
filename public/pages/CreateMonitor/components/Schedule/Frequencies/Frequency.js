/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { FormikSelect } from '../../../../../components/FormControls';
import { isInvalid, hasError } from '../../../../../utils/validate';

const frequencies = [
  { value: 'interval', text: 'By interval' },
  { value: 'daily', text: 'Daily' },
  { value: 'weekly', text: 'Weekly' },
  { value: 'monthly', text: 'Monthly' },
  { value: 'cronExpression', text: 'Custom cron expression' },
];

const flyoutFrequencies = [
  { value: 'interval', text: 'By interval' },
  { value: 'daily', text: 'Daily' },
  { value: 'weekly', text: 'Weekly' },
  { value: 'monthly', text: 'Monthly' },
];

const Frequency = ({ flyoutMode }) => (
  <FormikSelect
    name="frequency"
    formRow
    rowProps={{
      label: 'Frequency',
      isInvalid,
      error: hasError,
    }}
    inputProps={{
      options: flyoutMode ? flyoutFrequencies : frequencies,
      isInvalid,
      'data-test-subj': 'frequency_field',
    }}
  />
);

export default Frequency;
