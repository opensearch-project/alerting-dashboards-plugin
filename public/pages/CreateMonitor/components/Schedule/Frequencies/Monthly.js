/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

/*
 *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *   Licensed under the Apache License, Version 2.0 (the "License").
 *   You may not use this file except in compliance with the License.
 *   A copy of the License is located at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *   or in the "license" file accompanying this file. This file is distributed
 *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 *   express or implied. See the License for the specific language governing
 *   permissions and limitations under the License.
 */

import React, { Fragment } from 'react';
import { EuiFlexItem, EuiFlexGroup } from '@elastic/eui';

import Daily from './Daily';
import { FormikFieldNumber, FormikSelect } from '../../../../../components/FormControls';
import { isInvalid, hasError, validateMonthlyDay } from '../../../../../utils/validate';
import { monthlyTypes } from './utils/constants';

const Monthly = () => (
  <Fragment>
    <EuiFlexGroup alignItems="flexEnd" style={{ paddingLeft: '10px', marginTop: '5px' }}>
      <EuiFlexItem style={{ marginTop: '0px' }}>
        <FormikSelect
          name="monthly.type"
          formRow
          rowProps={{
            label: 'On the',
            isInvalid,
            error: hasError,
            style: { marginTop: '0px' },
          }}
          inputProps={{
            options: monthlyTypes,
            onChange: (e, field, form) => {
              form.setFieldValue('monthly.type', e.target.value);
            },
          }}
        />
      </EuiFlexItem>
      <EuiFlexItem style={{ marginTop: '0px' }}>
        <FormikFieldNumber
          name="monthly.day"
          formRow
          fieldProps={{ validate: validateMonthlyDay }}
          rowProps={{
            isInvalid,
            error: hasError,
            style: { marginTop: '0px' },
          }}
        />
      </EuiFlexItem>
    </EuiFlexGroup>
    <Daily />
  </Fragment>
);

export default Monthly;
