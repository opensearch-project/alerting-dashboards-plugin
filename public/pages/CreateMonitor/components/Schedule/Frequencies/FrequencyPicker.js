/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

/*
 *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *   Licensed under the Apache License, Version 2.0 (the "License").
 *   You may not use this file except in compliance with the License.
 *   A copy of the License is located at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *   or in the "license" file accompanying this file. This file is distributed
 *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 *   express or implied. See the License for the specific language governing
 *   permissions and limitations under the License.
 */

import React from 'react';
import { connect } from 'formik';

import CustomCron from './CustomCron';
import Daily from './Daily';
import Interval from './Interval';
import Monthly from './Monthly';
import Weekly from './Weekly';

const components = {
  daily: Daily,
  weekly: Weekly,
  monthly: Monthly,
  cronExpression: CustomCron,
  interval: Interval,
};

const FrequencyPicker = (props) => {
  const type = props.formik.values.frequency;
  const Component = components[type];
  return <Component />;
};

export default connect(FrequencyPicker);
