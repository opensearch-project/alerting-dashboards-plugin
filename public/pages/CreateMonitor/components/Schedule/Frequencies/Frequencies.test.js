/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { Formik } from 'formik';
import { render } from 'enzyme';

import { FORMIK_INITIAL_VALUES } from '../../../containers/CreateMonitor/utils/constants';
import Frequency from './Frequency';
import Interval from './Interval';
import Monthly from './Monthly';
import CustomCron from './CustomCron';
import FrequencyPicker from './FrequencyPicker';

describe('Frequencies', () => {
  test('renders Frequency', () => {
    const component = <Formik initialValues={FORMIK_INITIAL_VALUES} render={() => <Frequency />} />;

    expect(render(component)).toMatchSnapshot();
  });

  test('renders Interval', () => {
    const component = <Formik initialValues={FORMIK_INITIAL_VALUES} render={() => <Interval />} />;

    expect(render(component)).toMatchSnapshot();
  });

  test.skip('renders Monthly', () => {
    const component = <Formik initialValues={FORMIK_INITIAL_VALUES} render={() => <Monthly />} />;

    expect(render(component)).toMatchSnapshot();
  });

  test('renders CustomCron', () => {
    const component = (
      <Formik initialValues={FORMIK_INITIAL_VALUES} render={() => <CustomCron />} />
    );

    expect(render(component)).toMatchSnapshot();
  });

  test('renders FrequencyPicker', () => {
    const component = (
      <Formik initialValues={FORMIK_INITIAL_VALUES} render={() => <FrequencyPicker />} />
    );

    expect(render(component)).toMatchSnapshot();
  });
});
