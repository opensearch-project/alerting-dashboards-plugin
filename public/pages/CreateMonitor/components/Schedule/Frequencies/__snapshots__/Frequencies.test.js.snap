// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Frequencies renders CustomCron 1`] = `
<div
  class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionColumn euiFlexGroup--responsive"
  style="margin-top:5px"
>
  <div
    class="euiFlexItem"
    style="margin-top:0px"
  >
    <div
      class="euiFormRow"
      id="cronExpression-form-row-row"
      style="margin-top:0px"
    >
      <div
        class="euiFormRow__labelWrapper"
      >
        <label
          aria-invalid="false"
          class="euiFormLabel euiFormRow__label"
          for="cronExpression-form-row"
        >
          Run every
        </label>
      </div>
      <div
        class="euiFormRow__fieldWrapper"
      >
        <textarea
          class="euiTextArea euiTextArea--resizeVertical"
          data-test-subj="customCron_cronExpression_field"
          name="cronExpression"
          rows="6"
        >
          0 */1 * * *
        </textarea>
        <div
          class="euiFormHelpText euiFormRow__text"
          id="cronExpression-form-row-help-0"
        >
          <div
            class="euiText euiText--small"
          >
            Use 
            <a
              class="euiLink euiLink--primary"
              href="https://docs-beta.opensearch.org/monitoring-plugins/alerting/"
              rel="noopener noreferrer"
              target="_blank"
            >
              cron expressions
              <div>
                EuiIconMock
              </div>
              <span
                class="euiScreenReaderOnly"
              >
                (opens in a new tab or window)
              </span>
            </a>
             for complex schedules
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="euiFlexItem"
    style="margin-top:0px"
  >
    <div
      class="euiFormRow"
      id="timezone-form-row-row"
      style="margin-top:0px"
    >
      <div
        class="euiFormRow__fieldWrapper"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          class="euiComboBox"
          data-test-subj="timezoneComboBox"
          name="timezone"
          role="combobox"
        >
          <div
            class="euiFormControlLayout"
          >
            <div
              class="euiFormControlLayout__childrenWrapper"
            >
              <div
                class="euiComboBox__inputWrap euiComboBox__inputWrap--noWrap euiComboBox__inputWrap-isClearable"
                data-test-subj="comboBoxInput"
                tabindex="-1"
              >
                <p
                  class="euiComboBoxPlaceholder"
                >
                  Select a timezone
                </p>
                <div
                  class="euiComboBox__input"
                  style="font-size:14px;display:inline-block"
                >
                  <input
                    aria-controls=""
                    data-test-subj="comboBoxSearchInput"
                    id="timezone"
                    role="textbox"
                    style="box-sizing:content-box;width:1px"
                    value=""
                  />
                  <div
                    style="position:absolute;top:0;left:0;visibility:hidden;height:0;overflow:scroll;white-space:pre"
                  />
                </div>
              </div>
              <div
                class="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
              >
                <button
                  aria-label="Open list of options"
                  class="euiFormControlLayoutCustomIcon euiFormControlLayoutCustomIcon--clickable"
                  data-test-subj="comboBoxToggleListButton"
                  type="button"
                >
                  <div>
                    EuiIconMock
                  </div>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Frequencies renders Frequency 1`] = `
<div
  class="euiFormRow"
  id="frequency-form-row-row"
>
  <div
    class="euiFormRow__labelWrapper"
  >
    <label
      aria-invalid="false"
      class="euiFormLabel euiFormRow__label"
      for="frequency-form-row"
    >
      Frequency
    </label>
  </div>
  <div
    class="euiFormRow__fieldWrapper"
  >
    <div
      class="euiFormControlLayout"
    >
      <div
        class="euiFormControlLayout__childrenWrapper"
      >
        <select
          class="euiSelect"
          data-test-subj="frequency_field"
          id="frequency"
          name="frequency"
        >
          <option
            selected=""
            value="interval"
          >
            By interval
          </option>
          <option
            value="daily"
          >
            Daily
          </option>
          <option
            value="weekly"
          >
            Weekly
          </option>
          <option
            value="monthly"
          >
            Monthly
          </option>
          <option
            value="cronExpression"
          >
            Custom cron expression
          </option>
        </select>
        <div
          class="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
        >
          <span
            class="euiFormControlLayoutCustomIcon"
          >
            <div>
              EuiIconMock
            </div>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Frequencies renders FrequencyPicker 1`] = `
<div
  class="euiFlexGroup euiFlexGroup--gutterMedium euiFlexGroup--alignItemsFlexStart euiFlexGroup--directionRow euiFlexGroup--responsive"
>
  <div
    class="euiFlexItem"
  >
    <div
      class="euiFormRow"
      id="period.interval-form-row-row"
    >
      <div
        class="euiFormRow__labelWrapper"
      >
        <label
          aria-invalid="false"
          class="euiFormLabel euiFormRow__label"
          for="period.interval-form-row"
        >
          Run every
        </label>
      </div>
      <div
        class="euiFormRow__fieldWrapper"
      >
        <div
          class="euiFormControlLayout"
        >
          <div
            class="euiFormControlLayout__childrenWrapper"
          >
            <input
              class="euiFieldNumber euiFieldNumber--withIcon"
              data-test-subj="interval_interval_field"
              min="1"
              name="period.interval"
              type="number"
              value="1"
            />
            <div
              class="euiFormControlLayoutIcons"
            >
              <span
                class="euiFormControlLayoutCustomIcon"
              >
                <div>
                  EuiIconMock
                </div>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="euiFlexItem"
  >
    <div
      class="euiFormRow euiFormRow--hasEmptyLabelSpace"
      id="period.unit-form-row-row"
    >
      <div
        class="euiFormRow__fieldWrapper"
      >
        <div
          class="euiFormControlLayout"
        >
          <div
            class="euiFormControlLayout__childrenWrapper"
          >
            <select
              class="euiSelect"
              data-test-subj="interval_unit_field"
              id="period.unit"
              name="period.unit"
            >
              <option
                selected=""
                value="MINUTES"
              >
                Minute(s)
              </option>
              <option
                value="HOURS"
              >
                Hours
              </option>
              <option
                value="DAYS"
              >
                Days
              </option>
            </select>
            <div
              class="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
            >
              <span
                class="euiFormControlLayoutCustomIcon"
              >
                <div>
                  EuiIconMock
                </div>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Frequencies renders Interval 1`] = `
<div
  class="euiFlexGroup euiFlexGroup--gutterMedium euiFlexGroup--alignItemsFlexStart euiFlexGroup--directionRow euiFlexGroup--responsive"
>
  <div
    class="euiFlexItem"
  >
    <div
      class="euiFormRow"
      id="period.interval-form-row-row"
    >
      <div
        class="euiFormRow__labelWrapper"
      >
        <label
          aria-invalid="false"
          class="euiFormLabel euiFormRow__label"
          for="period.interval-form-row"
        >
          Run every
        </label>
      </div>
      <div
        class="euiFormRow__fieldWrapper"
      >
        <div
          class="euiFormControlLayout"
        >
          <div
            class="euiFormControlLayout__childrenWrapper"
          >
            <input
              class="euiFieldNumber euiFieldNumber--withIcon"
              data-test-subj="interval_interval_field"
              min="1"
              name="period.interval"
              type="number"
              value="1"
            />
            <div
              class="euiFormControlLayoutIcons"
            >
              <span
                class="euiFormControlLayoutCustomIcon"
              >
                <div>
                  EuiIconMock
                </div>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="euiFlexItem"
  >
    <div
      class="euiFormRow euiFormRow--hasEmptyLabelSpace"
      id="period.unit-form-row-row"
    >
      <div
        class="euiFormRow__fieldWrapper"
      >
        <div
          class="euiFormControlLayout"
        >
          <div
            class="euiFormControlLayout__childrenWrapper"
          >
            <select
              class="euiSelect"
              data-test-subj="interval_unit_field"
              id="period.unit"
              name="period.unit"
            >
              <option
                selected=""
                value="MINUTES"
              >
                Minute(s)
              </option>
              <option
                value="HOURS"
              >
                Hours
              </option>
              <option
                value="DAYS"
              >
                Days
              </option>
            </select>
            <div
              class="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
            >
              <span
                class="euiFormControlLayoutCustomIcon"
              >
                <div>
                  EuiIconMock
                </div>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
